STM32Cube/Tasks/controller.c

```diff
@@ -48,13 +42,13 @@ void controlTask(void *argument)
 		if(extension > CONTROLLER_MAX_EXTENSION) extension = CONTROLLER_MAX_EXTENSION;
 		if(extension < CONTROLLER_MIN_EXTENSION) extension = CONTROLLER_MIN_EXTENSION;
 
-		//TODO log to queue and send to CAN
 
-		//Test Code
-		char buffer[16] = {0};
-		sprintf(&buffer, "Target Ext: %d\r\n", (int)(extension * 1000));
-		//HAL_UART_Transmit(&huart4, &buffer, 16, 10); //Blocking API, 10ms timeout
-		osDelayUntil(CONTROLLER_DELAY_TICKS); //Implements periodic behaviour (nominal delay - time spent running the loop code)
+		if(extensionAllowed())
+		{
+			can_msg_t msg;
+			build_actuator_cmd_analog( (uint32_t) millis(), extension, ACTUATOR_AIRBRAKES_SERVO, &msg);
```

Comment:
The decleration of that function looks like this: 

`bool build_actuator_cmd_analog(uint32_t timestamp, enum ACTUATOR_ID actuator_id, float actuator_cmd, can_msg_t *output)`

So the order would be timestamp, then actuatorID, then extension, then msg object. I think you flipped actuatorID and extension.